graylog2-inputs/src/main/java/org/graylog2/inputs/codecs/GelfChunkAggregator.java

```diff
@@ -221,19 +224,25 @@ public boolean equals(Object o) {
 
             final ChunkEntry that = (ChunkEntry) o;
 
+            if (!id.equals(that.id)) return false;
             if (firstTimestamp != that.firstTimestamp) return false;
-            if (!chunkSlotsWritten.equals(that.chunkSlotsWritten)) return false;
-            //noinspection RedundantIfStatement
-            if (!payloadArray.equals(that.payloadArray)) return false;
+            if (chunkSlotsWritten.get() != that.chunkSlotsWritten.get()) return false;
+            if (!equalsAtomicReferenceArray(this.payloadArray, that.payloadArray)) return false;
```

Comment:
Specifically https://github.com/Graylog2/graylog2-server/blob/a0f396e330f45f1d78376796329eaad9a93cbc04/graylog2-inputs/src/main/java/org/graylog2/inputs/codecs/GelfChunkAggregator.java#L134-L138 and https://github.com/Graylog2/graylog2-server/blob/a0f396e330f45f1d78376796329eaad9a93cbc04/graylog2-inputs/src/main/java/org/graylog2/inputs/codecs/GelfChunkAggregator.java#L182 show the access pattern for eviction and do not care about the atomic fields at all.